### DISCUSSION TITLE
Geojson loaded as file or per API ?

### DISCUSSION THREAD

**Discussion by francoiscrouzet** [archaiodata/thacer#61] [Ideas]
@AlZorba 

On peut charger les fichiers par ajax
ou
on peut les inserer dans le projet en tant que variable js
ou
On peut les charger par une API

Par ailleurs, quand on les charges par ajax on peut les sauver en tant que variable dans le composant, ou à la idig, en local storage

L'idée de base est que les API c'est mieux, mais en fait on peut se poser la question. Pour Thacer, les données sont en lecture seule. Donc à priori elles ne changent pas ou tres rarement
Par ailleurs, elles ont une taille raisonnable, dans le sens où on peut tout charger en front
On se retrouve finalement avec une app hyper reactive, et qui pourrait presque marcher en offline

Il s'agirait de reflechir aux avantages inconvenients des differentes methodes
points clefs:
- taille de la 1ere charge VS charges consecutives
- mise en cache
- mode offline
- reactivité
- update des données

La problematique n'est pas clair pour moi, pensons-y, parlons-en !! :)

**Comment by AlZorba** [comment]
Loaded by API in ver 3.0.0 to be improved

**Reply by francoiscrouzet** [reply]
ah c'est vrai que tu as amelioré des choses, mais cette issue est beaucoup beaucoup plus generale pour moi. Il y a 5 ou 6 problematiques differentes qui ne sont pas claires
